### DISCUSSION TITLE
Language support

### DISCUSSION THREAD

**Discussion by LeonardoHSilva** [alandtse/alexa_media_player#2649] [Q&A] [answered]
Hi. Is it possible to have alexa tts messages in other languages? Something like this:

action: notify.alexa_media
data:
  message: A porta da garagem esteve aberta 10 minutos.
  title: Your Garage Door Friend
  target: media_player.leonardo_s_2nd_echo_dot
  data: 
    type: tts
    language: pt-PT

Thanks.

**Comment by neuquenfr** [comment, marked as answer]
Hello @LeonardoHSilva,

I use the following code:
=====
action: tts.google_say
data:
  cache: false
  entity_id: media_player.dot_del_escritorio
  language: "{{states ('input_select.msg_language')}}"
  message: "'{{states ('input_text.msg_wildcard')}}"
=====

The "tts.google_say" command works well with both Google and Alexa.
At each message I send to Alexa,
1. I place the language ("es", "en", "fr") in input_select.msg_language (from a SELECT button);
2. I use input_text helpers containing my fixed messages ("hola", "hello", "salut") of which the name ends with the 2 letters of the language code ("input_text.msg_blabla_es", "input_text.msg_blabla_en", "input_text.msg_labla_fr"). The container of the message is called "input_text.msg_wildcard" and I use the following code:
=====
action: input_text.set_value ### placing the content of the fixed message helper which matches the chosen language: "input_text.msg.blabla.es" (the "es" comes from "states) ("input_text.msg.blabla.en" would have returned the manually translated message in English, etc.).
metadata: {}
data:
  value: "{{ states('input_text.msg_tv_' ~ states('input_select.msg_language')) }}" ### the value of, for example, input_text.msg_blabla.es or  input_text.msg_blabla.en
target:
  entity_id: input_text.msg_wildcard ### the final message which will be spoken out by Alexa or Google Home.
=====
I hope it serves.
ps: sorry for the format of this message, it is the first time I publish an answer :-)
